### PR TITLE
automatically bump openssl's master commit hash

### PR DESCRIPTION
**Pull request by reaperhulk**

### CODE REVIEW

File: .github/workflows/openssl-version-bump.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: Bump OpenSSL version
+permissions:
+  contents: read
+
+on:
+  workflow_dispatch:
+  schedule:
+    # Run daily
+    - cron: "0 0 * * *"
```
Comment (alex) [inline .github/workflows/openssl-version-bump.yml:9]: This is going to produce PRs that basically always have conflicts with the boring ones, won't it?

Comment (alex) [reply .github/workflows/openssl-version-bump.yml:9]: Or make it all one job 😬

Comment (reaperhulk) [reply .github/workflows/openssl-version-bump.yml:9]: We could! could maybe even parametrize the job?

Comment (alex) [reply .github/workflows/openssl-version-bump.yml:9]: Parameterize would have the same 2x PR problem, right?

File: .github/workflows/boringssl-version-bump.yml [hunk 2]
```diff
@@ -14,37 +14,52 @@ jobs:
...
       - uses: actions/checkout@v3.1.0
-      - id: check-sha
+      - id: check-sha-boring
         run: |
           SHA=$(git ls-remote https://boringssl.googlesource.com/boringssl refs/heads/master | cut -f1)
+          LAST_COMMIT=$(grep boringssl .github/workflows/ci.yml | grep TYPE | grep -oE '[a-z0-9]{40}')
           if ! grep -q "$SHA" .github/workflows/ci.yml; then
-            echo "BORING_SHA=${SHA}" >> $GITHUB_OUTPUT
+            echo "COMMIT_SHA=${SHA}" >> $GITHUB_OUTPUT
+            echo "COMMIT_MSG=\"## BoringSSL\n[Commit: ${SHA}](https://boringssl.googlesource.com/boringssl/+/${COMMIT_SHA})\n\n[Diff](https://boringssl.googlesource.com/boringssl/+/${LAST_COMMIT}..${SHA}) between the last commit hash merged to this repository and the new commit.\"" >> $GITHUB_OUTPUT
```
Comment (alex) [inline .github/workflows/boringssl-version-bump.yml:23]:
```suggestion
            echo -e "COMMIT_MSG=\"## BoringSSL\n[Commit: ${SHA}](https://boringssl.googlesource.com/boringssl/+/${COMMIT_SHA})\n\n[Diff](https://boringssl.googlesource.com/boringssl/+/${LAST_COMMIT}..${SHA}) between the last commit hash merged to this repository and the new commit.\"" >> $GITHUB_OUTPUT
```

File: .github/workflows/boringssl-version-bump.yml [hunk 3]
```diff
@@ -14,37 +14,52 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3.1.0
-      - id: check-sha
+      - id: check-sha-boring
         run: |
           SHA=$(git ls-remote https://boringssl.googlesource.com/boringssl refs/heads/master | cut -f1)
+          LAST_COMMIT=$(grep boringssl .github/workflows/ci.yml | grep TYPE | grep -oE '[a-z0-9]{40}')
```
Comment (alex) [inline .github/workflows/boringssl-version-bump.yml:20]:
```suggestion
          LAST_COMMIT=$(grep boringssl .github/workflows/ci.yml | grep TYPE | grep -oE '[a-f0-9]{40}')
```

File: .github/workflows/boringssl-version-bump.yml [hunk 4]
```diff
@@ -14,37 +14,52 @@ jobs:
...
           if ! grep -q "$SHA" .github/workflows/ci.yml; then
-            echo "BORING_SHA=${SHA}" >> $GITHUB_OUTPUT
+            echo "COMMIT_SHA=${SHA}" >> $GITHUB_OUTPUT
+            echo -e "COMMIT_MSG=\"## BoringSSL\n[Commit: ${SHA}](https://boringssl.googlesource.com/boringssl/+/${COMMIT_SHA})\n\n[Diff](https://boringssl.googlesource.com/boringssl/+/${LAST_COMMIT}..${SHA}) between the last commit hash merged to this repository and the new commit.\"" >> $GITHUB_OUTPUT
+          fi
+      - id: check-sha-openssl
+        run: |
+          SHA=$(git ls-remote https://github.com/openssl/openssl refs/heads/master | cut -f1)
+          LAST_COMMIT=$(grep openssl .github/workflows/ci.yml | grep TYPE | grep -oE '[a-z0-9]{40}')
```
Comment (alex) [inline .github/workflows/boringssl-version-bump.yml:28]:
```suggestion
          LAST_COMMIT=$(grep openssl .github/workflows/ci.yml | grep TYPE | grep -oE '[a-f0-9]{40}')
```

File: .github/workflows/boringssl-version-bump.yml [hunk 5]
```diff
@@ -14,37 +14,52 @@ jobs:
...
+          if ! grep -q "$SHA" .github/workflows/ci.yml; then
+            echo "COMMIT_SHA=${SHA}" >> $GITHUB_OUTPUT
+            echo "COMMIT_MSG=\"## OpenSSL\n[Commit: ${SHA}](https://github.com/openssl/openssl/commit/${COMMIT_SHA})\n\n[Diff](https://github.com/openssl/openssl/compare/${LAST_COMMIT}..${SHA}) between the last commit hash merged to this repository and the new commit.\"" >> $GITHUB_OUTPUT
```
Comment (alex) [inline .github/workflows/boringssl-version-bump.yml:31]:
```suggestion
            echo -e "COMMIT_MSG=\"## OpenSSL\n[Commit: ${SHA}](https://github.com/openssl/openssl/commit/${COMMIT_SHA})\n\n[Diff](https://github.com/openssl/openssl/compare/${LAST_COMMIT}..${SHA}) between the last commit hash merged to this repository and the new commit.\"" >> $GITHUB_OUTPUT
```

**Review by alex — APPROVED** [review]

**Pull request merged**